### PR TITLE
upgrade mean of rotations to new LinearAlgebra syntax

### PR DESCRIPTION
**Pull request by colinxs**
mean was complaining about no eigfact function existing. This PR updates to current LinearAlgebra syntax which is eigvecs.

### CODE REVIEW

File: src/euler_types.jl [hunk 1]
```diff
@@ -14,7 +14,7 @@ for axis in [:X, :Y, :Z]
     @eval begin
         struct $RotType{T} <: Rotation{3,T}
             theta::T
-            $RotType{T}(theta) where {T} = new{T}(theta)
+            $RotType{T}(theta::Real) where {T} = new{T}(theta)
```
Comment (tkoolen) [inline src/euler_types.jl]: This change seems unrelated and would break e.g. interop with `SymPy.Sym`. Please revert.

Comment (colinxs) [reply src/euler_types.jl]: oops! haha I left that in by accident. There are some problems with mean() in its current form that I was trying to address. Namely that you cannot take the mean of two or three axis rotations. I'll file an issue for that!

Comment (colinxs) [reply src/euler_types.jl]: and I'll definitely add a test. I'm a little slammed right now but in a day or two!

**Comment by hyrodium** [comment]
I'll close this PR. See #177, #178.
